torch/testing/_internal/dist_utils.py

```diff
@@ -87,28 +88,42 @@ def new_test_method(self, *arg, **kwargs):
 def noop():
     pass
 
-def wait_until_node_failure(rank):
+def wait_until_node_failure(rank, expected_error_regex=".*"):
     '''
     Loops until an RPC to the given rank fails. This is used to
     indicate that the node has failed in unit tests.
+    Args:
+    rank (int) - Rank of the node expected to fail
+    expected_error_regex (optional, str): Regex of exception message expected. Useful to ensure a specific failure
+    occurs, not just any.
     '''
     while True:
         try:
             rpc.rpc_sync("worker{}".format(rank), noop, args=())
-            time.sleep(0.5)
-        except Exception:
-            break
+            time.sleep(0.1)
+        except Exception as e:
+            if re.match(pattern=expected_error_regex, string=str(e)):
```

Comment:
Was the intention to enforce the error message matches the given regex?

Reply:
Yes, this is to ensure that we get a desired error message when waiting for the node failure. My use case for this is that when testing failure edge cases, we are able to push out the RPC before the other node shuts down, and then get a timed out message instead of the expected error message due to the node being shutdown. In that case we will automatically retry here.